### PR TITLE
DPC-1466 Integrate New Relic

### PR DESCRIPTION
**Pull request by orenfromberg**
### Fixes [BCDA-1466](https://jira.cms.gov/browse/BCDA-1466)

This branch integrates the New Relic agent so that we can mark deployments as well as instrument our app for performance data.

### Proposed Changes

* initializing the new relic agent in SSAS

### Change Details

This branch doesn't do anything more than integrate the agent so that New Relic is aware of it.

### Security Implications

- [ ] new software dependencies

- [ ] security controls or supporting software altered

- [ ] new data stored or transmitted

- [ ] security checklist is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

- [x] no PHI/PII is affected by this change

### Acceptance Validation

- [x] Agent successfully registered with New Relic

image
image

### Feedback Requested

All feedback is welcome. Should we include any basic instrumentation in this PR or save for another ticket?

### CODE REVIEW

**Review by dhgreene — APPROVED** [review]
Is this paired with an ops PR?

**Comment by orenfromberg** [comment]
@dhgreene no, should it be? New Relic registers it automagically the first time it phones home.

**Review by CosmicSyntax — APPROVED** [review]

**Comment by msnook** [comment]
> @dhgreene no, should it be? New Relic registers it automagically the first time it phones home.

I don't think we'd need an ops change, outside of manually setting the env var in SSM.

File: ssas/service/main/main.go [hunk 1]
```diff
@@ -90,6 +90,16 @@ func init() {
 
 	// used by both `doResetSecret` and `doGetXdata`
 	flag.StringVar(&clientID, "client-id", "", "a system's client id")
+
+	appName := os.Getenv("NEW_RELIC_APP_NAME")
```
Comment (msnook) [inline ssas/service/main/main.go:94]: I'm surprised this wasn't already in place for ACO API.  What are the implications to ACO API of adding this?

Comment (orenfromberg) [reply ssas/service/main/main.go:94]: If the agent doesn't get initialized then the logger will warn about it but the app should continue merrily along. This is tested by the [build and test github actions](https://github.com/CMSgov/bcda-ssas-app/pull/89/checks?check_run_id=2973071386) which doesn't provide `NEW_RELIC_APP_NAME` or `NEW_RELIC_LICENSE_KEY` as secrets.

Comment (msnook) [reply ssas/service/main/main.go:94]: 🎉

**Review by msnook — APPROVED** [review]
👍

**Pull request merged**